PsfLauncher/main.cpp

```diff
@@ -71,185 +64,199 @@ int launcher_main(PWSTR args, int cmdShow) noexcept try
         throw_win32(ERROR_NOT_FOUND, "Error: could not find matching appid in config.json and appx manifest");
     }
 
-    auto exeName = appConfig->get("executable").as_string().wide();
     auto dirPtr = appConfig->try_get("workingDirectory");
-    auto dirStr = dirPtr ? dirPtr->as_string().wide() : nullptr;
-
+    auto dirStr = dirPtr ? dirPtr->as_string().wide() : L"";
     auto exeArgs = appConfig->try_get("arguments");
-    auto exeArgString = exeArgs ? exeArgs->as_string().wide() : (wchar_t*)L"";
-    auto monitor = PSFQueryAppMonitorConfig();
-
 
     // At least for now, configured launch paths are relative to the package root
     std::filesystem::path packageRoot = PSFQueryPackageRootPath();
-    auto exePath = packageRoot / exeName;
 
-    // Allow arguments to be specified in config.json now also.
-    std::wstring cmdLine = L"\"" + exePath.filename().native() + L"\" " + exeArgString + L" " + args;
+    ErrorInformation error;

```

Comment:
>ErrorInformation [](start = 4, length = 16)

For future code, consider an exception model where you can simply construct-and-throw the necessary bits of info.  Then you don't need to see the clutter of if checks around IsThereAnError, PSFReportError calls, and returns.  It all gets tucked away so you can focus on your business logic.

Reply:
Yeah, as I was in the middle of adding ErrorInformation, I realized that everything had to follow the "Check then throw" pattern.

Instead of doing a "Check and throw" do more something like, construct an ErrorInformation where the error is taking place and throwing there?